App/BraveWidgets/TopNewsWidget.swift

```diff
@@ -0,0 +1,241 @@
+        WidgetTopNewsView(entry: entry)
+      }
+    } else {
+      WidgetTopNewsView(entry: entry)
+    }
+  }
+}
+
+private struct LockScreenTopNewsView: View {
+  var entry: TopNewsEntry
+  
+  var body: some View {
+    if let topic = entry.topic {
+      VStack(alignment: .leading, spacing: 2) {
+        Text(topic.title)
+          .font(.system(size: 12, weight: .bold, design: .rounded))
+          .lineSpacing(0)
+          .layoutPriority(2)
+          .multilineTextAlignment(.leading)
+        HStack(spacing: 3) {
+          Image(braveSystemName: "brave.logo")
+            .foregroundColor(.orange)
+            .font(.system(size: 12))
+            .padding(.trailing, -1)
+          Divider()
+            .frame(height: 11)
+          Text("\(topic.publisherName)")
+            .lineLimit(1)
+            .font(.system(size: 11, weight: .semibold, design: .rounded))
+            .foregroundColor(.secondary)
+            .minimumScaleFactor(0.75)
+        }
+        .foregroundColor(.secondary)
+      }
+      .allowsTightening(true)
+      .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
+      .widgetURL(topic.url)
+    } else {
+      Text("")
+    }
+  }
+}
+
+private struct WidgetTopNewsView: View {
+  var entry: TopNewsEntry
+  
+  var body: some View {
+    if let topic = entry.topic {
+      VStack(alignment: .leading) {
+        HStack() {
+          Image(braveSystemName: "brave.logo")
+            .font(.footnote)
+            .imageScale(.large)
+            .foregroundColor(Color(.braveOrange))
+            .padding(4)
+            .background(Color(.white).clipShape(Circle()).shadow(color: .black.opacity(0.2), radius: 2, y: 1))
+        }
+        Spacer()
+        VStack(alignment: .leading, spacing: 4) {
+          Text(topic.title)
+            .shadow(radius: 2, y: 1)
+            .lineLimit(3)
+            .font(.system(size: 16, weight: .semibold, design: .rounded))
+            .foregroundColor(.white)
+            .allowsTightening(true)
+          Text(topic.publisherName)
+            .shadow(radius: 2, y: 1)
+            .lineLimit(1)
+            .minimumScaleFactor(0.7)
+            .font(.system(size: 11, weight: .medium, design: .rounded))
+            .foregroundColor(.white.opacity(0.95))
+        }
+      }
+      .padding()
+      .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .leading)
+      .background(
+        Group {
+          if let image = entry.image {
+            Image(uiImage: image)
+              .resizable()
+              .aspectRatio(contentMode: .fill)
+              .overlay(
+                LinearGradient(
+                  colors: [.black.opacity(0.0), .black.opacity(0.6)], startPoint: .top, endPoint: .bottom
+                )
+              )
+          } else {
+            LinearGradient(braveGradient: .darkGradient01)
+          }
+        }
+      )
+      .widgetURL(topic.url)
+    } else {
+      Text("Failed to load")
```

Comment:
Do we translate it, and is this error view good enough?

Reply:
We probably need a translation and maybe something nicer jhere